src/main.rs

```diff
@@ -248,6 +256,13 @@ async fn main() {
             })()
             .await;
 
+            // Incrementing a counter in an async context is surprisingly hard in Rust,
+            // so we don't know how many blobs we deleted (for now, we can count the log lines).
```

Comment:
great tip - I'll implement that in future PR

Reply:
I suppose by "surprisingly hard" I mean relative to JS, purely because JS is single-threaded and garbage collected so there's a whole bunch of stuff you don't need to worry about. The ease of the async/await syntax in Rust lends a false comparison

Reply:
Bit of a more advanced topic but...

There's also[ `AtomicU64`](https://doc.rust-lang.org/std/sync/atomic/struct.AtomicU64.html) which can be shared directly. It will be a faster since it uses the CPUs internal atomic CAS instructions rather than a Mutex.

You'll probably want [`fetch_add`](https://doc.rust-lang.org/std/sync/atomic/struct.AtomicU64.html#method.fetch_add)

There's also this library which exposes a nicer API if all you want to do is bump an integer up. It's basically just a wrapper around the above. Relaxed memory ordering is probably fine in this case.

https://docs.rs/atomic-counter/latest/atomic_counter/